helix-core/src/main/java/org/apache/helix/HelixManagerProperty.java

```diff
@@ -55,12 +58,13 @@ public HelixManagerProperty(Properties helixManagerProperties, CloudConfig cloud
   }
 
   private HelixManagerProperty(String version, long healthReportLatency,
-      HelixCloudProperty helixCloudProperty,
+      HelixCloudProperty helixCloudProperty, InstanceConfig defaultInstanceConfig,
```

Comment:
I prefer not use InstanceConfig. Because InstanceConfig requires you to have an ID for this InstanceConfig object.

But when we merge, it is hard for us to determine whether the ID will be overridden or not as the ID is mandatory field. 

And this field does matter for ZNode creation.